### PR TITLE
Create my enumerable

### PR DESCRIPTION
**Pull request by edwardoluyara**
Here in this project  I have worked solo due to the responses I received from [@Marco Lizza](https://mail.google.com/mail/u/0/#inbox/FMfcgzGmvLNNNGvfbpNfZwdCtWQWdjnC) but aside from that I have been able to complete the following requirements:

- Created a class `MyList` that has an instance variable `@list`.
- And in `MyList` I have  implemented a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require_relative './my_Enumerable'
```
Comment (Dipeshtwis) [inline my_list.rb]: - [ ] There is a typo in this line. it should be `require_relative './my_enumerable'`. Please fix this.

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class MyList
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&number)
+    @list.each(&number)
+  end
```
Comment (Dipeshtwis) [inline my_list.rb]: - [ ] Inside `each` method you have to implement your logic to behave like each. kindly implement this.

Like this:

```
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

File: README.md [hunk 3]
```diff
@@ -1 +1,39 @@
...
+Create a class `MyList` that has an instance variable `@list`.
+In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.
+
+...you can see more of the requirements [here](https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/create_your_own_enumerable.md)
+
+## Built with
+Programming language: `Ruby`
+Frameworks: `NaN`
+Technology: `NaN`
+Testing tool: [Rubocup](https://rubocop.org/)
```
Comment (Dipeshtwis) [inline README.md]: - [ ] I would recommend including the steps to clone and run the project locally to make the README professional.

```
git clone https://github.com/oluyaratosin123/My-Enumerable.git
cd My-Enumerable
ruby my_list.rb
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @oluyaratosin123, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
### Hi @oluyaratosin123, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**